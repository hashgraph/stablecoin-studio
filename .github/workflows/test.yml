name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test