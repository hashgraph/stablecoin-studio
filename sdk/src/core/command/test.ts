/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
