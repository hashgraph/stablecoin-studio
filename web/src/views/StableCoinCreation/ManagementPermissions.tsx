import { Heading, HStack, Stack, Text, VStack } from '@chakra-ui/react';
import type { Control, FieldValues } from 'react-hook-form';
import { useWatch } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import SwitchController from '../../components/Form/SwitchController';
import KeySelector from './components/KeySelector';

interface ManagementPermissionsProps {
	control: Control<FieldValues>;
}

const ManagementPermissions = (props: ManagementPermissionsProps) => {
	const { control } = props;
	const { t } = useTranslation(['global', 'stableCoinCreation']);

	const isManagementPermissions = useWatch({
		control,
		name: 'managementPermissions',
	});

	const keys = [
		{
			name: 'adminKey',
			nameTranslate: t('stableCoinCreation:managementPermissions.admin'),
		},
		{
			name: 'supplyKey',
			nameTranslate: t('stableCoinCreation:managementPermissions.supply'),
		},
		{
			name: 'wipeKey',
			nameTranslate: t('stableCoinCreation:managementPermissions.wipe'),
		},
		{
			name: 'freezeKey',
			nameTranslate: t('stableCoinCreation:managementPermissions.freeze'),
		},
		{
			name: 'pauseKey',
			nameTranslate: t('stableCoinCreation:managementPermissions.pause'),
		},
	];

	return (
		<VStack h='full' justify={'space-between'} pt='80px'>
			<Stack minW={400}>
				<Heading
					data-testid='title'
					fontSize='16px'
					fontWeight='600'
					mb={10}
					lineHeight='15.2px'
					textAlign={'left'}
				>
					{t('stableCoinCreation:managementPermissions.title')}
				</Heading>
				<Stack as='form' spacing={6} pb={6}>
					<HStack mb={4}>
						<Text maxW={'252px'} fontSize='14px' fontWeight='400' lineHeight='17px'>
							{t('stableCoinCreation:managementPermissions.description')}
						</Text>
						<SwitchController
							control={control}
							name={'managementPermissions'}
							defaultValue={true}
						/>
					</HStack>
					{isManagementPermissions === false ? (
						<>
							{keys.map((item) => {
								return (
									<KeySelector
										key={item.name}
										control={control}
										name={item.name}
										label={item.nameTranslate}
									/>
								);
							})}
						</>
					) : (
						<Stack spacing={2} fontSize='14px' fontWeight='400'>
							{keys.map((item, index) => {
								return <Text key={index}>{item.nameTranslate}</Text>;
							})}
						</Stack>
					)}
				</Stack>
			</Stack>
		</VStack>
	);
};

export default ManagementPermissions;
