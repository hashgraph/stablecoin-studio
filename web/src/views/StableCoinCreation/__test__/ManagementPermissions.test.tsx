import { FieldValues, useForm } from 'react-hook-form';
import { render } from '../../../test/index';
import translations from '../../../translations/en/stableCoinCreation.json';
import ManagementPermissions from '../ManagementPermissions';
import { CreateRequest } from '@hashgraph-dev/stablecoin-npm-sdk';

jest.mock('react-hook-form', () => ({
	...jest.requireActual('react-hook-form'),
	Controller: () => <></>,
	useWatch: () => true,
	useForm: () => ({
		control: () => ({}),
		watch: () => jest.fn(),
		setValue: () => jest.fn(),
	}),
}));

const form = useForm<FieldValues>({
	mode: 'onChange',
});

const { control, watch, setValue } = form;
const request = new CreateRequest({
	name: '',
	symbol: '',
	decimals: 6,
	createReserve: false,
});

describe(`<${ManagementPermissions.name} />`, () => {
	test.skip('should render correctly', () => {
		const component = render(
			<ManagementPermissions
				control={control}
				request={request}
				watch={watch}
				setValue={setValue}
			/>,
		);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has title', () => {
		const component = render(
			<ManagementPermissions
				control={control}
				request={request}
				watch={watch}
				setValue={setValue}
			/>,
		);

		const header = component.getByTestId('title');
		expect(header).toHaveTextContent(translations.managementPermissions.keysTitle);
	});
});
