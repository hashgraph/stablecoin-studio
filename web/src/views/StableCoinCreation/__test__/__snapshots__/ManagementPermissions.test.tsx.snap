// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManagementPermissions /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-130jr3z"
  >
    <div
      class="chakra-stack css-135a0es"
    >
      <h2
        class="chakra-heading css-p7jgmy"
        data-testid="title"
      >
        Permission management
      </h2>
      <form
        class="chakra-stack css-1tjpxl5"
      >
        <div
          class="chakra-stack css-1tchtwa"
        >
          <p
            class="chakra-text css-3crm2z"
          >
            Assign default values
          </p>
        </div>
        <div
          class="chakra-stack css-oat9fj"
        >
          <p
            class="chakra-text css-mgupmj"
          >
            Wipe - The smart contract
          </p>
          <p
            class="chakra-text css-mgupmj"
          >
            Freeze - The smart contract
          </p>
          <p
            class="chakra-text css-mgupmj"
          >
            Pause - The smart contract
          </p>
        </div>
        <div
          class="chakra-stack css-1tchtwa"
        >
          <p
            class="chakra-text css-3crm2z"
          >
            Do you want to enable KYC?
          </p>
        </div>
        <div
          class="chakra-stack css-owjkmg"
        />
        <div
          class="chakra-stack css-1tchtwa"
        >
          <p
            class="chakra-text css-3crm2z"
          >
            Do you want to add custom fees?
          </p>
        </div>
        <div
          class="chakra-stack css-owjkmg"
        />
      </form>
    </div>
    <div
      class="chakra-stack css-135a0es"
    >
      <div
        class="chakra-accordion css-0"
      >
        <div
          class="chakra-accordion__item css-17mg6aq"
        >
          <button
            aria-controls="accordion-panel-:r0:"
            aria-expanded="false"
            class="chakra-accordion__button css-uttm9k"
            data-index="0"
            id="accordion-button-:r0:"
            type="button"
          >
            <span
              class="css-1irdfqk"
            >
              Roles assignment
            </span>
            <svg
              aria-hidden="true"
              class="chakra-icon chakra-accordion__icon css-186l2rg"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </button>
          <div
            class="chakra-collapse"
            style="overflow: hidden; display: none; opacity: 0; height: 0px;"
          >
            <div
              aria-labelledby="accordion-button-:r0:"
              class="chakra-accordion__panel css-1wt3bp2"
              id="accordion-panel-:r0:"
              role="region"
            >
              <div
                class="css-0"
                data-testid="supplier-quantity"
              >
                <div
                  class="chakra-stack css-owjkmg"
                />
                <div
                  class="chakra-stack css-1wytzre"
                >
                  <p
                    class="chakra-text css-4dzt3e"
                  >
                    Unlimited?
                  </p>
                </div>
              </div>
              <div
                class="chakra-stack css-owjkmg"
              />
              <div
                class="chakra-stack css-owjkmg"
              />
              <div
                class="chakra-stack css-owjkmg"
              />
              <div
                class="chakra-stack css-owjkmg"
              />
              <div
                class="chakra-stack css-owjkmg"
              />
              <div
                class="chakra-stack css-owjkmg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
